### ISSUE TITLE
docstring-missing-returns (`DOC201`) reports on `__new__`

### ISSUE THREAD

**Issue by tjkuson** [astral-sh/ruff#13079]
Running `ruff check --select DOC201 --preview --isolated` on

```python
class Foo:
    def __new__(cls, x: int) -> "Foo":
        """A very helpful docstring.

        Args:
            x: An integer.
        """
        ...  # Do something with x.
        return super().__new__(cls)
```

reports a docstring-missing-returns (DOC201) diagnostic.

The expected behaviour is that the return value of a `__new__` method is not expected to be documented as it simply returns an instance of the class.

Related to #8085

Search terms: DOC201, docstring-missing-returns, `__new__`
